### ISSUE TITLE
 -emb_file could match -emb_file_enc, -emb_file_dec

### ISSUE THREAD

**Issue by pinzhenliao** [OpenNMT/OpenNMT-py#702]
Hi,
Thank you very much for the great tool.

When I followed the instruction on page http://opennmt.net/OpenNMT-py/FAQ.html#how-do-i-use-pretrained-embeddings-e-g-glove to **use Pretrained embeddings (e.g. GloVe),** 

when I run 

./tools/embeddings_to_torch.py -emb_file "glove_dir/glove.6B.300d.txt"  -dict_file "data/demo.vocab.pt"  -output_file "data/embeddings"

I got this error:

**embeddings_to_torch.py: error: ambiguous option: -emb_file could match -emb_file_enc, -emb_file_dec**

You might want to update the FAQ page.

**Comment by pltrdy** [comment]
The doc is actually out of date. One must explicitely specify which vector file you want to use, for both the encoder and the decoder. 

If you want to use the same vectors, just set:
`./tools/embeddings_to_torch.py -emb_file_enc "glove_dir/glove.6B.300d.txt" -emb_file_dec "glove_dir/glove.6B.300d.txt" -dict_file "data/demo.vocab.pt" -output_file "data/embeddings"`

**Comment by pltrdy** [comment]
@wingsyuan indeed.

**Comment by wingsyuan** [comment]
Thanks very much!
Here is my another questions:

* enc: 20925 match, 8793 missing, (70.41%)
	* dec: 20923 match, 13342 missing, (61.06%)

Filtered embeddings:
 	* enc: torch.Size([29718, 300])
	* dec: torch.Size([34265, 200])
how does the system handle the missing embedding? Are the misssing parts tokens under the random initialization in the beginning? then both the miss and unmiss are update when trainning?

what does the Filtered embeddings mean? they are seem to of different dim.. Can I use them directly? 

thanks very much for your help!

**Comment by pltrdy** [comment]
I just checked, missing tokens are in the vocabulary and the embeddings. They get 0 valued vector, which is maybe not a super good idea, random would have been better I guess.

By default, embedding weights **will** be updated during training, you can set `-fix_word_vecs_enc` / `-fix_word_vecs_dec` to keep (all) vectors from updating.

The embeddings dimensions are `[vocabulary_size x vec_dim]`, the `vec_dim` depends on your embedding file. You probably gave a 200 dim src emb and 300 dim tgt emb.


----
Where default value is set: https://github.com/OpenNMT/OpenNMT-py/blob/master/tools/embeddings_to_torch.py#L70

**Comment by wingsyuan** [comment]
@pltrdy  Thank you very much for your help!

Can I just add two lines below into the commands to train?

python  train.py -data /data/test1 -save_model /data/model1 \
 
	**-pre_word_vecs_enc "data/embeddings.enc.pt" \
        -pre_word_vecs_dec "data/embeddings.dec.pt" \**     

        -layers 6 -rnn_size 512 -word_vec_size 512 -transformer_ff 2048 -heads 8  \
 
        -encoder_type transformer -decoder_type transformer -position_encoding \
 
        -train_steps 200000  -max_generator_batches 2 -dropout 0.1 \
 
        -batch_size 4096 -batch_type tokens -normalization tokens  -accum_count 2 \
 
        -optim adam -adam_beta2 0.998 -decay_method noam -warmup_steps 8000 -learning_rate 2 \
 
        -max_grad_norm 0 -param_init 0  -param_init_glorot \
 
        -label_smoothing 0.1 -valid_steps 10000 -save_checkpoint_steps 10000 \
 
        -world_size 2 -gpu_ranks 0 1

Are  two files "embeddings.enc.pt" and "embeddings.dec.pt"  contain the word embedding of the src and tgt vocabulary and have the same word order as them(the src vocabulay and tgt vocabulary)?

**Comment by pltrdy** [comment]
Yup, it should be ok. 

The word order isn't the same tho, i.e. one word is associated to a token (i.e. integer value) based on the vocabulary (basically index in a list of words). A given word may, or may not, have different tokens in src vocab and tgt vocab.
